client/client-lib/src/mint/backup.rs

```diff
@@ -143,6 +157,118 @@ impl MintClient {
             .await?;
         Ok(())
     }
+
+    /// Fetch epochs in a given range and send them over `sender`
+    ///
+    /// Since WASM's `spawn` does not support join handles, we indicate
+    /// errors via `sender` itself.
+    ///
+    /// TODO: could be internal to recovery_loop?
+    async fn fetch_epochs(
+        &self,
+        epoch_range: RangeInclusive<u64>,
+        sender: mpsc::Sender<api::Result<EpochHistory>>,
+        task_handle: &TaskHandle,
+    ) {
```

Comment:
This looks like a use case for [`unfold`](https://docs.rs/futures/0.3.25/futures/stream/fn.unfold.html) that would allow you to create a stream over epochs.

Reply:
My understanding is that streams don't really work in parallel. They only work like iterators, which means the fetching will only start when the consumer needs it. With tasks, the epochs are pre-fetched while the consumer is processing them. In the future we might even consume epochs from different members at the same time to share the load and parallize it. If necessary of course, not sure what will be the bottleneck here.